blog/tests/test_views.py

```diff
@@ -0,0 +1,106 @@
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [post, past_post])
+            self.assertTemplateUsed(response, 'blog/post_list.html')
+            self.assertContains(response, post)
+            self.assertContains(response, past_post)
+            self.assertNotContains(response, future_post)
+        with patch('django.utils.timezone.now', lambda: datetime(day=1, month=4, year=3016, tzinfo=tz)):
+            response = self.client.get(reverse('post_list'))
+            self.assertListEqual(list(response.context['posts']), [future_post, post, past_post])
+            response = self.client.get(reverse('post_list'))
+            self.assertEqual(200, response.status_code)
+
+    def test_detail_view(self):
+        """Testing detail page when post is not exist and when it exists."""
+        response = self.client.get(reverse('post_detail', kwargs={'pk': 1}))
+        self.assertEqual(404, response.status_code)
+        post = Post.objects.create(author=self.user, title='Test', text='superText')
+        response = self.client.get(reverse('post_detail', kwargs={'pk': post.pk}))
+        self.assertEqual(200, response.status_code)
+        self.assertContains(response, post)
+
+    def test_post_new_view(self):  # so only the tests with login are broken
+        """Testing new post view before and after login."""
+        response = self.client.get(reverse('post_new'))
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username=self.USERNAME, password=self.PASSWORD)
+        self.assertTrue(authorization)
+        response = self.client.post(reverse('post_new'), {'author': self.user, 'title': 'Test', 'text': 'superText', },
+                                    follow=True)
+        self.assertEqual(200, response.status_code)
+        self.assertContains(response, 'Test')
+        self.assertRedirects(response, reverse('post_detail', kwargs={'pk': 1}))
+
+    def test_post_edit(self):  # and this one
+        """Testing edit views before and after user login."""
+        response = self.client.get(reverse('post_edit', kwargs={'pk': 1}))
+        self.assertEqual(302, response.status_code)
+        authorization = self.client.login(username=self.USERNAME, password=self.PASSWORD)
+        self.assertTrue(authorization)
+        response = self.client.get(reverse('post_edit', kwargs={'pk': 1}))
+        # This time we'are logged but post is not exist
+        self.assertEqual(404, response.status_code)
+        post = Post.objects.create(author=self.user, title='Test', text='superText')
+        response = self.client.get(reverse('post_edit', kwargs={'pk': post.pk}))
+        self.assertEqual(200, response.status_code)
+        response = self.client.post(reverse('post_edit', kwargs={'pk': 1}), {'author': self.user, 'title': 'Test',
```

Comment:
in fact, you don't have to change author and title when testing change of contents only.